frontend/src/components/Canvas/CanvasDraw/CanvasDraw.tsx

```diff
@@ -62,12 +62,13 @@ const CanvasDraw: React.FC<Props> = ({
 }) => {
   const [color, setColor] = useState<DrawingColor>(DrawingColor.BLACK);
   const [brushType, setBrushType] = useState<BrushType>(BrushType.THIN);
-  const [isPainting, setIsPainting] = useState(false);
   const drawing = useRef<Paint>([]);
   const undoneDrawing = useRef<Paint>([]);
-  const [currentLine, setCurrentLine] = useState<Line | null>(null);
-  const [mousePosition, setMousePosition] = useState<Point | undefined>(undefined);
+  const mousePosition = useRef<Point | null>(null);
+  const currentLine = useRef<Line | null>(null);
```

Comment:
pk en fait les ref c'est mieux que les state ?

Reply:
je croyais avoir répondu mais ça a disparu :

Les refs c'est une variable mutable attachée au composant. Je peux à tout moment y accéder et la setter, de tous les endroits où je la passe, sans avoir besoin de faire un setVar() qui va rerender tout le composant. 

A la base je voulais mettre la variable mousePosition directement sur la window et pas comme state, bah la ref c'est un peu pareil, sauf que c'est quand même attaché au composant. Mais quand tu l'updates ça déclenche pas un rerendering.

Reply:
Si on avait un composant classe, on aurait juste pu setter et lire `this.mousePosition` du composant.

Reply:
Pourquoi on utilises pas des refs partout alors au lieu  des state ?

Reply:
Parce que des fois tu veux que ça rerender. Typiquement, si ton state est affiché à un endroit dans ta vue, il faut que React rerender quand ton state a changé.